### PR TITLE
JAMES-3680 Allow to disable SMTP plain auth

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/netty/SMTPServer.java [hunk 1]
```diff
@@ -86,13 +86,17 @@ public static AuthenticationAnnounceMode parse(String authRequiredString) {
...
+
+
         public static AuthenticationConfiguration parse(HierarchicalConfiguration<ImmutableNode> configuration) {
             return new AuthenticationConfiguration(
                 Optional.ofNullable(configuration.getString("auth.announce", null))
                     .map(AuthenticationAnnounceMode::parse)
                     .orElseGet(() -> fallbackAuthenticationAnnounceMode(configuration)),
-                Optional.ofNullable(configuration.getBoolean("auth.requireSSL", null))
-                    .orElse(false));
+            Optional.ofNullable(configuration.getBoolean("auth.requireSSL", null))
```
Comment (vttranlina) [inline server/protocols/protocols-smtp/src/main/java/org/apache/james/smtpserver/netty/SMTPServer.java:96]: configuration.getBoolean("auth.requireSSL", false),
                configuration.getBoolean("auth.plainAuthEnabled", true));

**Comment by Arsnael** [comment]
Rebased and applied @vttranlina comment :)

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebased on stable PR #782

**Review by vttranlina — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebased on master and squashed

**Pull request merged**